apisix/plugins/jwt-auth.lua

```diff
@@ -200,15 +209,28 @@ local function fetch_jwt_token(conf, ctx)
         return token
     end
 
-    token = ctx.var["arg_" .. conf.query]
+    local uri_args = core.request.get_uri_args(ctx) or {}
+    token = uri_args[conf.query]
     if token then
+        if conf.hide_credentials then
+            -- hide for query
+            uri_args[conf.query] = nil
+            core.request.set_uri_args(ctx, uri_args)
+        end
         return token
     end
 
     local val = ctx.var["cookie_" .. conf.cookie]
     if not val then
         return nil, "JWT not found in cookie"
     end
+
+    if conf.hide_credentials then
+        -- hide for cookie
+        local reset_val = conf.cookie .. "=deleted; Max-Age=0"
```

Comment:
The Max-Age attribute is for Set-Cookie. What about parsing the Cookie header and removing the relative field?

Reply:
Do you mean handling multiple header cookie fields like this?
Downstream:
`"headers":[ "Cookie": "jwt-cookie=some-credentials; foo=bar" ]`
Upstream:
`"headers":[ "Cookie": "foo=bar" ]`

Reply:
Yes

Reply:
Thanks, I will rewrite it.